### PR TITLE
depend on local diff-highlight.

### LINKED ISSUE #6: Hard dependency on diff-highlight from diff-so-fancy
Maybe we should just depend on `diff-highlight` from inside of `diff-so-fancy`.

That would mean people instead configure it with `git diff | diff-so-fancy` directly.

If `diff-highlight` isn't found, we could just pass through the `stdin` and perhaps issue a warning that says "hey you're missing diff-highlight bud"

Seems a little more user-friendly. wdyt?

### PR DESCRIPTION
**Pull request by paulirish**
fixes #6 

hows this look to everyone?

### CODE REVIEW

**Comment by paulirish** [comment]
@noscript what do you think?

**Comment by aziz** [comment]
:+1: Hiding the complexity from end user.

**Comment by stevemao** [comment]
This looks great :+1:

**Comment by noscript** [comment]
I think it's better to check first if there is already `diff-highlight` in `$PATH` and if not then fallback to the one in `./third_party`.

**Comment by stevemao** [comment]
@noscript since we are adding `update-deps.sh` its more likely that our `diff-highlight` is probably more updated.
Also if we only hard depend on our own `diff-highlight` and we have more control. Say there might be a mismatch of `diff-highlight` on user's machine and we have to ask user manually fix their dependency.

**Comment by noscript** [comment]
Ignore my previous comment, that won't work. Version of `diff-highlight` that comes with `git` package has no executable flag.

> @noscript since we are adding update-deps.sh its more likely that our diff-highlight is probably more updated.

A side issue is that before this PR `diff-so-fancy` didn't have an internal dependency on `diff-highlight`. One could simply drop `diff-so-fancy` anywhere in $PATH and configure git pager.

After this PR it becomes less convenient for non-npm users, since `diff-so-fancy` will be looking for hardcoded `./third_party` directory.

This PR can be merged as is now and later I can introduce a flag for `diff-so-fancy` to fallback to old behavior.

File: update-deps.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+curl -o "third_party/diff-highlight/diff-highlight" "https://raw.githubusercontent.com/git/git/master/contrib/diff-highlight/diff-highlight"
+curl -o "third_party/diff-highlight/README" "https://raw.githubusercontent.com/git/git/master/contrib/diff-highlight/README"
```
Comment (stevemao) [inline update-deps.sh]: Can we just copy the whole folder or use glob instead of `curl`ing twice?

Comment (noscript) [inline update-deps.sh]: Not sure about `curl` but with `wget` we can do
```
wget -N -P third_party/diff-highlight https://raw.githubusercontent.com/git/git/master/contrib/diff-highlight/{diff-highlight,README}
```

**Comment by stevemao** [comment]
> A side issue is that before this PR diff-so-fancy didn't have an internal dependency on diff-highlight.

This is not true. If the npm bin takes higher precedence over user's own path that contains `diff-highlight`, user will always run our `diff-highlight`. if the other way, user will never run our own `diff-highlight`.



> later I can introduce a flag for diff-so-fancy to fallback

Like I said there might be a mismatch. `npm` uses local dependencies over global to avoid that. `gem` uses the global approach and I heard a lot of people having trouble. If we do introduce such a flag it'll be only for advanced users and we won't support that (Say user has v1 and our script depends on v2, its too much overhead to make it work on both for too little gain).

And, like @paulirish already mentioned, diff-highlight doesn't update very often so I wonder it's worth the effort...

File: update-deps.sh (hunk 1 above)
Comment (paulirish) [inline update-deps.sh]: wget isnt on mac by default, so i hesitate to use that.

**Comment by cht8687** [comment]
This PR not only changes the repo to a shell project but also clarifies the third-party dependency, quite good move!!! :+1:

**Pull request merged**